test/unittests/validation_test.cpp

```diff
@@ -129,6 +129,170 @@ TEST(validation, memory_size_no_memory)
         parse(wasm), validation_error, "memory instructions require imported or defined memory");
 }
 
+TEST(validation, store8_alignment)
```

Comment:
I think it is fine for `i32.store8` (and similarly for other instructions) to have biggest passing (1) and smallest failing (2) aligns. So 2 cases per instruction.